### ISSUE TITLE
Is it possible to add an image from a url to a slide?

### ISSUE THREAD

**Issue by Santiagoq3** [singerla/pptx-automizer#131]
something like this: 

ModifyImageHelper.setImage('https://your-amazon-s3-link-to-image.com/image.jpg').

thanks!

**Comment by singerla** [comment]
This is still a little experimental, please refer to https://github.com/singerla/pptx-automizer/blob/main/__tests__/generate-pptxgenjs-image.test.ts and your url at:

https://github.com/singerla/pptx-automizer/blob/f7a2d8055ed01fabf03100777d30843f0adf1b75/__tests__/generate-pptxgenjs-image.test.ts#L18

**Comment by Santiagoq3** [comment]
@singerla  thank you so much for your answer. I'm trying to do the same thing but after trying to use code from that test I get an error.

Trying to add an image based of an url from Amazon S3:
```ts
const automizer = new Automizer({
  templateDir: `/mnt/c/work`,
  templateFallbackDir: `/mnt/c/work`,
  outputDir: `/mnt/c/work/output`,
  useCreationIds: false,
  autoImportSlideMasters: false,
  removeExistingSlides: false,
  cleanup: false,
  compression: 0,
});

const url = "https://publics3bucket.com/test.pptx";
const response = await fetch(url);
const buffer = await response.arrayBuffer();
const bytes = new Uint8Array(buffer);

let pres = automizer.loadRoot(bytes)
pres.load(bytes, 'shapes');

pres.addSlide('shapes', 1, (slide) => {
  slide.generate((pptxGenJSSlide, objectName) => {
    pptxGenJSSlide.addImage({
      path: "https://publics3bucket.com/test.png",
      x: 1,
      y: 2,
      objectName,
    });
  });
});
```

Error obtained in console:
```
[Nest] 2040  - 11/15/2024, 3:37:23 PM   ERROR [ExceptionsHandler] slide.generate is not a function
TypeError: slide.generate is not a function
    at <anonymous> (/backend/src/pedido/pedido.service.ts:290:13)
    at Automizer.addSlide (/backend/node_modules/pptx-automizer/src/automizer.ts:345:7)
    at PedidoService.testPower (/backend/src/pedido/pedido.service.ts:288:10)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at PedidoController.postPowerPoint (/backend/src/pedido/pedido.controller.ts:195:17)
    at async /backend/node_modules/@nestjs/core/router/router-execution-context.js:46:28
    at async /backend/node_modules/@nestjs/core/router/router-proxy.js:9:17
```
Also:
```
Property 'generate' does not exist on type 'ISlide'.ts(2339)
```
should I use the pptxgenjs library? ```pptxgenjs```

These are the versions of dependencied I thought would be relevant to know:
```
pptx-automizer: "^0.5.2"
node: v20.15.0
```

**Comment by singerla** [comment]
I'm sorry, you need to clone repo & yarn link. This is not yet published

Am 15. November 2024 20:26:40 MEZ schrieb Santiago Quipildor ***@***.***>:
>@singerla  thank you so much for your answer. I'm trying to do the same thing but after trying to use code from that test I get an error.
>
>Trying to add an image based of an url from Amazon S3:
>```ts
>const automizer = new Automizer({
>  templateDir: `/mnt/c/work`,
>  templateFallbackDir: `/mnt/c/work`,
>  outputDir: `/mnt/c/work/output`,
>  useCreationIds: false,
>  autoImportSlideMasters: false,
>  removeExistingSlides: false,
>  cleanup: false,
>  compression: 0,
>});
>
>const url = "https://publics3bucket.com/test.pptx";
>const response = await fetch(url);
>const buffer = await response.arrayBuffer();
>const bytes = new Uint8Array(buffer);
>
>let pres = automizer.loadRoot(bytes)
>pres.load(bytes, 'shapes');
>
>pres.addSlide('shapes', 1, (slide) => {
>  slide.generate((pptxGenJSSlide, objectName) => {
>    pptxGenJSSlide.addImage({
>      path: "https://publics3bucket.com/test.png",
>      x: 1,
>      y: 2,
>      objectName,
>    });
>  });
>});
>```
>
>Error obtained in console:
>```
>[Nest] 2040  - 11/15/2024, 3:37:23 PM   ERROR [ExceptionsHandler] slide.generate is not a function
>TypeError: slide.generate is not a function
>    at <anonymous> (/backend/src/pedido/pedido.service.ts:290:13)
>    at Automizer.addSlide (/backend/node_modules/pptx-automizer/src/automizer.ts:345:7)
>    at PedidoService.testPower (/backend/src/pedido/pedido.service.ts:288:10)
>    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
>    at PedidoController.postPowerPoint (/backend/src/pedido/pedido.controller.ts:195:17)
>    at async ***@***.***/core/router/router-execution-context.js:46:28
>    at async ***@***.***/core/router/router-proxy.js:9:17
>```
>Also:
>```
>Property 'generate' does not exist on type 'ISlide'.ts(2339)
>```
>should I use the pptxgenjs library? ```pptxgenjs```
>
>These are the versions of dependencied I thought would be relevant to know:
>```
>pptx-automizer: "^0.5.2"
>node: v20.15.0
>```
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/singerla/pptx-automizer/issues/131#issuecomment-2479770577
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by singerla** [comment]
This is available now with v0.6.0!
